src/uberComponents/ResourceListPage/hooks.ts

```diff
@@ -120,3 +120,22 @@ function extractPrimaryResourcesFactory<R extends Resource>(resourceType: R['res
             .map((entry) => entry.resource as R);
     };
 }
+
+export function useSearchBarForGenericFilters(getFilters?: ResourceListProps<Resource>['getFilters']) {
+    const defaultValues = Object.fromEntries(getFilters?.({}).map((column) => [column.id, column.defaultValue]) ?? []);
+    const [values, setValues] = useState(defaultValues);
+    const onResetFilters = useCallback(() => {
+        setValues(defaultValues);
+    }, [defaultValues]);
+
+    const columnDefinitions = getFilters?.(values) ?? [];
+    // TODO: add validation
```

Comment:
Don't merge it until it's fixed!!!

And check reset filters - it should reset references